feat(jobs): implement job post update endpoint and add comprehensive tests

Daggahh opened the pull request:
## Description
Implements the PATCH endpoint `/api/v1/jobs/{id}` to allow users to update their job posts. Added comprehensive validation, authorization checks, and test coverage.

### Changes Made
- Added PATCH endpoint for updating job posts
- Implemented user authorization checks
- Added validation for update data
- Created comprehensive test suite
- Added error handling for invalid requests
- Maintained data integrity during updates
- Added SuperAdminGuard back to update endpoint
- Modified guards to allow both super admin and job owner to update jobs
- Removed docker-compose.yml from git tracking
- Updated tests to reflect new authorization logic


### Testing
✅ Added test cases for:
- Successful job updates
- Invalid data validation
- Empty update payload
- Authorization checks
- UUID validation
- Missing user authentication
- Data integrity preservation

### API Documentation
```typescript
PATCH /api/v1/jobs/{id}
Authorization: Bearer <token>
Content-Type: application/json

// Request body (all fields optional)
{
  "title": "Updated Job Title",
  "description": "Updated description",
  "salary_range": "100k_to_150k",
  "deadline": "2024-12-31T23:59:59Z"
}

// Success Response (200)
{
  "status": "success",
  "status_code": 200,
  "message": "Job updated successfully",
  "data": {
    // Updated job object
  }
}
```

### Checklist
- [x] Implementation follows existing patterns
- [x] Added comprehensive tests
- [x] Added proper error handling
- [x] Updated API documentation
- [x] Code follows project style guide
- [x] All tests passing

### Related Issue
Closes #1247

Daggahh committed:
feat(jobs, auth): implement job update endpoint & fix super admin creation (#1247)

- Add PATCH /api/v1/jobs/{id} endpoint for updating job posts
- Implement authorization for job owners and admins
- Fix super admin role assignment in user creation
- Ensure comprehensive validation and error handling
- Update API documentation with new endpoints
- Maintain data integrity during job updates
- Add test suite covering edge cases for job updates

This update introduces job post modification while ensuring proper authorization and data integrity.
It also resolves issues with super admin role creation and keeps API responses consistent.

Resolves #1247

phoenixdahdev commented on docker-compose.yml:
add this file to git ignore it is not needed

phoenixdahdev commented on src/modules/jobs/jobs.controller.ts:
this is open to both owner and super admin

phoenixdahdev commented on src/modules/jobs/jobs.controller.ts:
```diff
@@ -57,7 +62,7 @@ export class JobsController {
   }
 
-  @UseGuards(SuperAdminGuard)
+  @UseGuards(AuthGuard)
```
both super admin and owner

phoenixdahdev commented on src/modules/jobs/jobs.controller.ts:
```diff
@@ -104,6 +109,20 @@ export class JobsController {
   }
 
+  @Patch('/:id')
+  @UseGuards(AuthGuard, JobOwnerGuard)
```
add super admin guard back

Daggahh replied:
so making it 3 guards then? i thought the admin guard, covered for super admin too

Daggahh committed:
feat(auth): allow both super admin and job owner to update jobs

- Add SuperAdminGuard back to job update endpoint
- Allow both super admin and job owner to update jobs
- Remove docker-compose.yml from git tracking

Resolves #1257

Daggahh committed:
chore: resolve gitignore conflicts

- Remove duplicate entries
- Merge docker-compose and data directory ignores
- Clean up environment file ignores

Daggahh committed:
chore: resolve gitignore merge conflicts

- Merge duplicate entries
- Standardize data directory ignore
- Maintain all necessary ignore patterns

phoenixdahdev committed:
Merge branch 'dev' into dev
